src/ploomber/spec/taskspec.py

```diff
@@ -429,6 +447,8 @@ def _init_product(task_dict, meta, task_class, root_path, lazy_import):
     """
     product_raw = task_dict.pop('product')
 
+    _validate_product_extension(product_raw)
```

Comment:
please move this call to here: https://github.com/ploomber/ploomber/blob/6e7ae7ca155749f4efaf0093b7f0d0df462de924/src/ploomber/spec/taskspec.py#L72

this is where we determine if the user wants to dump data, hence, here we check the .csv/.parquet thing - because there are cases when the user has a SQL script but wants to create a new database table instead of dumping the data locally

so we need something like:

```python
        if extension == '.sql' and _safe_suffix(product) in {
                '.csv', '.parquet'
        }:
            return tasks.SQLDump
        else _extension_typo(_safe_suffix(product)):
           raise DAGSpecInitializationError('csb isnt a valid extension, did you mean csv?')
```